rali/rali/source/tf_meta_data_reader_detection.cpp

```diff
@@ -0,0 +1,279 @@
+    {
+        auto image_name = image_names[i];
+        auto it = _map_content.find(image_name);
+	
+        if(_map_content.end() == it){
+
+            BoundingBoxCords bb_coords;
+            BoundingBoxLabels bb_labels;
+            BoundingBoxCord box;
+
+            box.x = box.y = box.w = box.h = 0;
+            bb_coords.push_back(box);
+            bb_labels.push_back(0);
+            // bb_coords={};
+
+	    _output->get_bb_cords_batch()[i] = bb_coords;
+	    _output->get_bb_labels_batch()[i] = bb_labels;
+        }
+	else{
+        _output->get_bb_cords_batch()[i] = it->second->get_bb_cords();
+        _output->get_bb_labels_batch()[i] = it->second->get_bb_labels();
+	}
+    }
+
+
+}
+
+void TFMetaDataReaderDetection::print_map_contents()
+{
+
+    BoundingBoxCords bb_coords;
+    BoundingBoxLabels bb_labels;
+
+    std::cerr << "\nMap contents: \n";
+    for (auto& elem : _map_content) {
+        std::cerr << "Name :\t " << elem.first;
+        bb_coords = elem.second->get_bb_cords() ;
+        bb_labels = elem.second->get_bb_labels();
+        std::cerr << "\nsize of the element  : "<< bb_coords.size() << std::endl;
+        for(unsigned int i = 0; i < bb_coords.size(); i++){
+            std::cerr << " x : " << bb_coords[i].x << " y: :" << bb_coords[i].y << " width : " << bb_coords[i].w << " height: :" << bb_coords[i].h << std::endl;
+            std::cerr  << "Label Id : " << bb_labels[i] << std::endl;
+        }
+    }
+}
+
+void TFMetaDataReaderDetection::read_record(std::ifstream &file_contents, uint file_size, std::vector<std::string> &_image_name,
+    std::string user_label_key, std::string user_text_key, 
+    std::string user_xmin_key, std::string user_ymin_key, std::string user_xmax_key, std::string user_ymax_key,
+    std::string user_filename_key)
+{
+    // std::cerr << "The user_label_key is " << user_label_key << ", and the user_filename_key is " << user_filename_key << "\n";
+    // std::cerr << "user_xmin_key is " << user_xmin_key << ", user_ymin_key is " << user_ymin_key << "\n";
+    // std::cerr << "user_xmax_key is " << user_xmax_key << ", user_ymax_key is " << user_ymax_key << "\n";
+    uint length;
+    length = file_contents.tellg();
+
+    std::string temp;
+    size_t uint64_size, uint32_size;
+    uint64_t data_length;
+    uint32_t length_crc, data_crc;
+    uint64_size = sizeof(uint64_t); 
+    uint32_size = sizeof(uint32_t); 
+    char * header_length = new char [uint64_size];
+    char * header_crc = new char [uint32_size];
+    char * footer_crc = new char [uint32_size];
+    file_contents.read(header_length, uint64_size);
+    file_contents.read(header_crc, uint32_size);
+    memcpy(&data_length, header_length, sizeof(data_length));
+    memcpy(&length_crc, header_crc, sizeof(length_crc));
+
+    if(length + data_length + 16 == file_size){
+        _last_rec = true;
+    }
+    char *data = new char[data_length];
+    file_contents.read(data,data_length);
```

Comment:
Codacy found an issue: [Check buffer boundaries if used in a loop including recursive loops (CWE-120, CWE-20).  ](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5976200)